File: examples/03_module/stock/stock.py
# Lean Code — BSD 3-Clause License — Vivian Voss, 2026
# Scope: Read and check stock availability.


def read_stock_for_product(product):
    try:
        lines = open("stock.txt").read().strip().split("\n")
    except FileNotFoundError:
        return 0

    for line in lines:
        name, count = line.split(",", 1)
        if name == product:
            return int(count)

    return 0


def check_stock_availability(product, quantity):
    stock = read_stock_for_product(product)
    return stock >= quantity

File: examples/03_module/stock/stock.js
// Lean Code — BSD 3-Clause License — Vivian Voss, 2026
// Scope: Read and check stock availability.

import { readFileSync } from "node:fs";

function readStockForProduct(product) {
  const lines = readFileSync("stock.txt", "utf-8").trim().split("\n");

  for (const line of lines) {
    const [name, count] = line.split(",", 2);
    if (name === product) {
      return parseInt(count, 10);
    }
  }

  return 0;
}

function checkStockAvailability(product, quantity) {
  const stock = readStockForProduct(product);
  return stock >= quantity;
}

export { readStockForProduct, checkStockAvailability };
